readthedocs/builds/models.py

```diff
@@ -18,6 +18,7 @@
                                             REPO_TYPE_HG, GITHUB_URL,
                                             GITHUB_REGEXS, BITBUCKET_URL,
                                             BITBUCKET_REGEXS)
+from readthedocs.projects.tasks import clear_artifacts
```

Comment:
I think this introduced a cyclic dependency: https://travis-ci.org/rtfd/readthedocs.org/jobs/81886399#L213
You might want to move the import directly into the `delete` method.